pkg/controller/nstemplatetier/nstemplatetier_controller.go

```diff
@@ -0,0 +1,236 @@
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	activeTemplateUpdateRequests, err := r.activeTemplateUpdateRequests(instance)
+	if err != nil {
+		log.Error(err, "unable to get active TemplateUpdateRequests")
+		return reconcile.Result{}, err
+	}
+
+	members := r.retrieveMemberClusters()
+loop:
+	for _, member := range members {
+		// create a TemplateUpdateRequest if active count < max pol size
+		if activeTemplateUpdateRequests < maxPoolSize {
+			// find a MasterUserRecord which is not already up-to-date
+			// and for which there is no TemplateUpdateRequest yet
+			// fetch by subsets of "maxPoolSize + 1" size until a MasterUserRecord candidate is found
+			murs := toolchainv1alpha1.MasterUserRecordList{}
+			matchingLabels, err := newLabelSelector(instance, *member)
+			if err != nil {
+				log.Error(err, "unable to get MasterUserRecords to update")
+				return reconcile.Result{}, err
+			}
+			r.client.List(context.Background(), &murs,
+				client.InNamespace(instance.Namespace),
+				client.Limit(maxPoolSize+1),
+				matchingLabels,
+			)
+			log.Info("listed MasterUserRecords", "count", len(murs.Items), "cluster", member.Name)
+			for _, mur := range murs.Items {
+				for _, account := range mur.Spec.UserAccounts {
+					accountTmpls := account.Spec.NSTemplateSet
+					log.Info("checking mur account", "name", mur.Name, "tier", accountTmpls.TierName)
+					if accountTmpls.TierName != instance.Name {
+						// skip if the user account is not based on the current Tier
+						continue
+					}
+					// compare all TemplateRefs in the current MUR.UserAccount vs NSTemplateTier instance
+					if !reflect.DeepEqual(templateRefs(accountTmpls), templateRefs(instance.Spec)) {
+						log.Info("creating a TemplateUpdateRequest to update the MasterUserRecord", "name", mur.Name, "tier", instance.Name)
+						r.client.Create(context.TODO(), &toolchainv1alpha1.TemplateUpdateRequest{
+							ObjectMeta: metav1.ObjectMeta{
+								Name: mur.Name,
+								Labels: map[string]string{
+									toolchainv1alpha1.NSTemplateTierNameLabelKey: instance.Name,
+								},
+							},
+							Spec: toolchainv1alpha1.TemplateUpdateRequestSpec{
+								TierName:           instance.Name,
+								NSTemplateTierSpec: instance.Spec,
+							},
+						})
+						break loop // create a single TemplateUpdateRequest resource per reconcile loop
+					}
+				}
+			}
+		}
+	}
+	log.Info("done with creating TemplateUpdateRequest resources after update of NSTemplateTier", "tier", instance.Name)
+	return reconcile.Result{}, nil
+}
+
+func newLabelSelector(instance toolchainv1alpha1.NSTemplateTier, member cluster.FedCluster) (client.MatchingLabelsSelector, error) {
+	// compute the hash of the `.spec.namespaces[].templateRef` + `.spec.clusteResource.TemplateRef`
+	hash, err := computeTemplateRefsHash(instance)
+	if err != nil {
+		log.Error(err, "unable to generate the hash for the NSTemplateTier")
```

Comment:
Here and everywhere else in the controller, let's use `logger` from `Reconcile()` instead of `log`.

Reply:
ah right, of course 🤦‍♂️

Reply:
done in d0458e8